src/dmd/semantic3.d

```diff
@@ -407,7 +407,8 @@ private extern(C++) final class Semantic3Visitor : Visitor
                     sc2.insert(_arguments);
                     _arguments.parent = funcdecl;
                 }
-                if (f.linkage == LINK.d || f.parameterList.length)
+                if ((f.linkage == LINK.d || f.parameterList.length) &&
+                    !(sc.flags & SCOPE.Cfile))  // don't want to require importing stdarg for C files
```

Comment:
I was considering just adding a new entry to `VarArg` to distinguish it from normal variadic functions.